redux/longestStreakSlice.ts

```diff
@@ -22,36 +24,50 @@ export enum STAGE {
 function initializeGameState(): GameBlockState[] {
   let dummyArray: GameBlockState[] = [];
   for (let i = 0; i <= 20; i++) {
-    let x = getRndInteger(1, 9);
-    let y = getRndInteger(1, 9);
+    let x = getRndInteger(1, 10);
+    let y = getRndInteger(1, 10);
     let product: number = x * y;
     let productString: string = x + " x " + y;
 
     let initiateBlockState: GameBlockState = {
       text: product.toString(),
       value: product,
+      isProduct: true,
       state: BlockState.NOT_SELECTED,
     };
     dummyArray.push(initiateBlockState);
 
     initiateBlockState = {
       text: productString,
       value: product,
+      isProduct: false,
       state: BlockState.NOT_SELECTED,
     };
     dummyArray.push(initiateBlockState);
   }
 
-  //shuffle list
   dummyArray = shuffle(dummyArray);
+
   return dummyArray;
 }
 const initialState: LongestStreakState = {
   stage: STAGE.SET_RULES,
+  reset: false,
+  blocks: initializeGameState(),
+  handlePlayerSelect: 0,
+  isPlayerSelecting: false,
+  currentlySelectedBlock: null,
+  playerName: "Player 1",
+};
+
+const resetInitialState: LongestStreakState = {
+  stage: STAGE.PLAY_GAME,
+  reset: false,
```

Comment:
Why is this property needed?

Reply:
When is it used? When is it set to true?

Reply:
It allows me to re-initialize the game board with redux when I hit the reset button on the play_game stage. It gives me new numbers, etc.

Reply:
I had to clear state for the game in order to truly reset. Otherwise, if I just tried to initialize it would remember some parts of the state.